### DISCUSSION TITLE
Callbacks works once than stops

### DISCUSSION THREAD

**Discussion by ElBourino** [pschatzmann/ESP32-A2DP#569] [Q&A] [answered]
Hello,

Again thank you for this library..
I am facing an issue using callbacks, specifically the one to get playstatus (avrc_rn_playstatus_callback and avrc_rn_audio_state_callback). when called for the first time, they react as expected, but then it seems all callbacks are blocked after.

I am using a code based on one of the examples:
```
#include "BluetoothA2DPSink.h"

BluetoothA2DPSink a2dp_sink;

void avrc_rn_play_pos_callback(uint32_t play_pos) {
  Serial.printf("Play position is %d (%d seconds)\n", play_pos, (int)round(play_pos/1000.0));
}

void avrc_rn_playstatus_callback(esp_avrc_playback_stat_t playback) {
  switch (playback) {
    case esp_avrc_playback_stat_t::ESP_AVRC_PLAYBACK_STOPPED:
      Serial.println("Stopped.");
      break;
    case esp_avrc_playback_stat_t::ESP_AVRC_PLAYBACK_PLAYING:
      Serial.println("Playing.");
      break;
    case esp_avrc_playback_stat_t::ESP_AVRC_PLAYBACK_PAUSED:
      Serial.println("Paused.");
      break;
    case esp_avrc_playback_stat_t::ESP_AVRC_PLAYBACK_FWD_SEEK:
      Serial.println("Forward seek.");
      break;
    case esp_avrc_playback_stat_t::ESP_AVRC_PLAYBACK_REV_SEEK:
      Serial.println("Reverse seek.");
      break;
    case esp_avrc_playback_stat_t::ESP_AVRC_PLAYBACK_ERROR:
      Serial.println("Error.");
      break;
    default:
      Serial.printf("Got unknown playback status %d\n", playback);
  }
}

void setup() {
  Serial.begin(115200);
  a2dp_sink.set_avrc_rn_playstatus_callback(avrc_rn_playstatus_callback);
  a2dp_sink.set_avrc_rn_play_pos_callback(avrc_rn_play_pos_callback);
  a2dp_sink.start("MyMusic");
}

void loop() {
  Serial.println("Play or pause music to test callbacks.");
  delay(10000);
}
```

When uploaded, the monitor sees the first playback event but that is it. 
The chip doesn't seem blocked as the "Play or pause music to test callbacks." keeps showing up regularly.

Would you have an idea on why that is?
I am on a ESP32-WROOM-32-4M chip.

Thanks!

**Comment by pschatzmann** [comment, marked as answer]
I think you forgot to mention what ESP32 core version you are using and what exectly you are trying to do.
Are you sure that you would not want to use [this functionality](https://github.com/pschatzmann/ESP32-A2DP/blob/main/examples/bt_music_receiver_playing_status_callbacks/bt_music_receiver_playing_status_callbacks.ino) ?

In addition I could not reproduce your issue: Maybe you you need to describe the exact steps to reproduce it.  I was hitting start and stop on my Android player application several times and I was receiving the expected events. So it might be also an issue with your Android application ?

I was testing with ESP32 2.0.17!

ps. Dont  forget to set the log level to info, so that you can see what events you receive from your phone...
